taca/flowcell_parser/classes.py

```diff
@@ -0,0 +1,322 @@
+
+import os
+import csv
+import xml.etree.ElementTree as ET
+import logging
+
+from bs4 import BeautifulSoup #html parser
+
+class XTenParser(object):
+    def __init__(self, path):
+        if os.path.exists(path):
+            self.log=logging.getLogger('XTenParser')
+            self.path=path
+            self.parse()
+            self.create_db_obj()
+        else:
+            raise os.error("XTen flowcell cannot be found at {0}".format(path))
+
+    def parse(self):
+        fc_name=self.path[-9:]
+        rinfo_path=os.path.join(self.path, 'RunInfo.xml')
+        rpar_path=os.path.join(self.path, 'runParameters.xml')
+        ss_path=os.path.join(self.path, 'SampleSheet.csv')
+        lb_path=os.path.join(self.path, 'Demultiplexing', 'Reports', 'html', fc_name, 'all', 'all', 'all', 'laneBarcode.html')
+
+        try:
+            self.runinfo=XTenRunInfoParser(rinfo_path)
+        except OSError as e:
+            self.log.info(str(e))
+            self.runinfo=None
+        try:
+            self.runparameters=XTenRunParametersParser(rpar_path)
+        except OSError as e:
+            self.log.info(str(e))
+            self.runParameters=None
+        try:
+            self.samplesheet=XTenSampleSheetParser(ss_path)
+        except OSError as e:
+            self.log.info(str(e))
+            self.samplesheet=None
+        try:
+            self.lanebarcodes=XTenLaneBarcodeParser(lb_path)
+        except OSError as e:
+            self.log.info(str(e))
+            self.lanebarcodes=None
+
+
+    def create_db_obj(self):
+        self.obj={}
+        bits=os.path.basename(self.path).split('_')
+        name="{0}_{1}".format(bits[0], bits[-1])
+        self.obj['name']=name
+        if self.runinfo:
+            self.obj['RunInfo']=self.runinfo.data
+            if self.runinfo.recipe:
+                self.obj['run_setup']=self.runinfo.recipe
+        if self.runparameters:
+            self.obj.update(self.runparameters.data)
```

Comment:
any reason for updating the dict in this case instead of not adding a new key like with the other files?